docs/error-handling.md

```diff
@@ -0,0 +1,242 @@
+  if (params.namespaceId && params.packageId) {
+    const { dapper } = getLoaderTools();
+    try {
+      const readme = await dapper.getPackageReadme(
+        params.namespaceId,
+        params.packageId
+      );
+      return { readme };
+    } catch (error) {
+      handleLoaderError(error);
+    }
+  }
+  // ...
+}
+```
+
+-   **Source**: [`apps/cyberstorm-remix/cyberstorm/utils/errors/handleLoaderError.ts`](../../apps/cyberstorm-remix/cyberstorm/utils/errors/handleLoaderError.ts)
+-   **Example**: [`apps/cyberstorm-remix/app/p/tabs/Readme/Readme.tsx`](../../apps/cyberstorm-remix/app/p/tabs/Readme/Readme.tsx)
+
+### Error Handling in `loader` vs. `clientLoader`
+
+There is a notable difference in how errors are handled between `loader` and `clientLoader` functions.
+
+-   **`loader` (Server-side):** Errors are consistently processed through `handleLoaderError`. This utility catches errors from API calls (e.g., from Dapper), maps them to structured `UserFacingError` payloads, and throws a `Response`. This ensures that errors are standardized before they reach the UI, allowing for consistent presentation in the route's `ErrorBoundary`.
+
+-   **`clientLoader` (Client-side):** Error handling is less standardized. `clientLoader` functions often do not use `handleLoaderError`. Instead, they might:
+    -   Throw a generic `Error` object.
+    -   Have no explicit `try...catch` block, letting promise rejections from API calls be caught by the nearest Remix error boundary.
+
+This can lead to less specific error messages for the user. The recommendation is to apply the same `try...catch` and `handleLoaderError` pattern in `clientLoader` as is used in `loader` to ensure a consistent user experience.
+
+#### Bad Example (`clientLoader` without `handleLoaderError`)
+
+```typescript
+// This is not recommended
+export async function clientLoader({ params }: ClientLoaderFunctionArgs) {
+  const { dapper } = getLoaderTools();
+  try {
+    const team = await dapper.getTeam(params.teamId);
+    // Awaiting all follow-up data forces the UI to block on the clientLoader.
+    const members = await dapper.getTeamMembers(params.teamId);
+    return { team, members };
+  } catch (error) {
+    // Errors will bubble to Remix but we miss the user-facing payload mapping.
+    throw error;
+  }
+}
+```
+
+#### Good Example (`clientLoader` with `handleLoaderError`)
+
+```typescript
+// This is the recommended approach
+export async function clientLoader({ params }: ClientLoaderFunctionArgs) {
+  const { dapper } = getLoaderTools();
+
+  let team;
+  try {
+    // Await the first fetch only because the follow-up request depends on it.
+    team = await dapper.getTeam(params.teamId);
+  } catch (error) {
+    // This provides a consistent, user-facing error.
+    handleLoaderError(error);
+  }
+
+  // Kick off dependent work without awaiting so Suspense can resolve it downstream.
+  const members = dapper.getTeamMembers(team.slug);
```

Comment:
Should this be inside the try-block above? If it's intentionally left out, maybe use the comment above to describe what happens if it's erroneously placed inside the try-block.